enhancements/user-workload-monitoring.md

```diff
@@ -0,0 +1,270 @@
+---
+title: user-workload-monitoring
+authors:
+  - "@s-urbaniak"
+  - "@brancz"
+reviewers:
+  - "@sichvoge"
+  - "@openshift/openshift-team-monitoring"
+approvers:
+  - "@sichvoge"
+  - "@openshift/openshift-team-monitoring"
+creation-date: 2019-09-19
+last-updated: 2019-09-19
+status: implementable
+---
+
+# user-workload-monitoring
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+This enhancement describes a new feature for OpenShift cluster observability, namely user workload monitoring.
+It is an extension to the existing cluster monitoring stack to enable observability of user namespaces.
+Currently, the cluster monitoring stack only reconciles OpenShift namespaces.
+
+## Motivation
+
+Currently, there is no supported solution for customers to monitor their business critical applications.
+The existing cluster monitoring stack has no support for adding scrape targets in user namespaces
+as it is only available for OpenShift components scraping workloads from `openshift-*` namespaces.
+Other Red Hat products, outside the OpenShift framework components also
+don’t have a monitoring solution to integrate their metrics and out-of-the-box insights into their well being.
+Currently it is difficult to achieve metering/chargeback for Red Hat products without a central observability stack.
+
+### Goals
+
+Allow deployment and reconciliation of scrape targets, recording rules, and alerting rules in user namespaces.
+This effectively means the following custom resources:
+
+-  [ServiceMonitor](https://github.com/openshift/prometheus-operator/blob/master/Documentation/api.md#servicemonitor)
+- [PodMonitor](https://github.com/openshift/prometheus-operator/blob/master/Documentation/api.md#podmonitor)
+- [PrometheusRule](https://github.com/openshift/prometheus-operator/blob/master/Documentation/api.md#prometheusrule)
+
+Allow the correlation between app and cluster centric metrics.
+
+### Non-Goals
+
+Out of scope are:
+
+- Definition of Dashboards.
+- Deployment of Prometheus instances in user namespaces.
```

Comment:
This is still possible by default in 4.2, is this stating you won’t add *more* capability?

Reply:
Deployment of Prometheus instances in 4.2 is necessary as we don't provide one for user workload monitoring ourselves currently. User workload monitoring is meant to be mutually exclusive with an OLM based prometheus operator deployment. I will add some more clarifying statements.

Reply:
Are you working with any particular product team on this requirement? I am hoping he AppSRE team needing Prometheus deployment for Telemeter is customer zero?

Reply:
@jeremyeder we've been discussing this with @aditya-konarde at the monitoring f2f. It is true, that this feature here won't help AppSRE, @brancz is going to have a deployment plan for prometheus operator for telemetry purposes.